### PR TITLE
Rebuild YOLOFastestV2

### PR DESCRIPTION
**Pull request by hglee98**
## Description

This PR introduces `YOLOFastestV2`, an improved version of the YOLOFastest object detection model.

As the NP-Trainer already presents `YOLOXLoss` including SimOTA matching algorithm, I use the idea within `YOLOFastestLoss`. (Bonus: we can also make YOLOFastest to be an anchor-free when we modify model configuration!). The pretrained-weights aren't ready yet. So, I will share the weight as soon as possible when the fine-tuning process is done.

Related to #376 

We recommend to link at least one existing issue for PR. Before your create a PR, please check if there is an issue for this change.  

## Change(s)

- New Bounding Box Decoding Formula. we propose a new formula for decoding bounding box width and height:
```math
\begin{align}
b_w &= 2 * p_w * \left(\tanh\left(\frac{t_w}{2}-0.549306\right)+1\right) \\
b_h &= 2 * p_h * \left(\tanh\left(\frac{t_h}{2}-0.549306\right)+1\right) \\
\end{align}
```
- Loss Function Enhancement. Incorporated ideas from `YOLOXLoss` (including SimOTA matching algorithm) into `YOLOFastestLoss` by inheritance.
- Potential for the novel Anchor-Free YOLOFastest. With minor modifications to the model configuration, YOLOFastest can now be transformed into an anchor-free model.

TODOs
- [x] Complete weight porting via fine-tuning bbox_reg head.

## Code Formatting

If you PR to either `master` or `dev` branch, you should follow the code linting process. Please check your code with `lint_check.sh` in `./scripts` directory.
For more information, please read the contribution guide in `CONTRIBUTING.md`. 

## Changelog

If your PR is granted to `dev` branch, codeowner will add a brief summary of the change to the **Upcoming Release** section of the [`CHANGELOG.md`](https://github.com/Nota-NetsPresso/netspresso-trainer/blob/master/CHANGELOG.md) file including a link to the PR (formatted in markdown) and a link to your github profile.

For example,

```
- Added a new feature by `@myusername` in [PR 2023](https://github.com/Nota-NetsPresso/netspresso-trainer/pull/2023)
```

Please enable **Allow edits and access to secrets by maintainers** so that our maintainers can update the `CHANGELOG.md`.

### CODE REVIEW

**Review by illian01 — APPROVED** [review]
Thank you for your efforts.
I updated configuration files, weights file url, and documentation. other things look good to me.
